Sources/Sentry/SentryHttpTransport.m

```diff
@@ -75,8 +79,10 @@ - (void)sendEnvelope:(SentryEnvelope *)envelope
         return;
     }
 
-    [self.fileManager storeEnvelope:envelope];
-    [self sendAllCachedEnvelopes];
+    [self.dispatchQueue dispatchAsyncWithBlock:^{
```

Comment:
Even though not relevant here, it's worth noting in most SDKs this introduces the challenge of capturing a message when the process is about to crash.
This is when `flush` enters. So the caller would have a way to block the calling thread (even if it's the UI thread) while this block runs and the event is actually written to disk, or sent to Sentry.

But since we have SentryCrash here which is a separate mechanism, worst case is tthat we lose a `captureX` call by user code because of a hard crash which is acceptable for the trade off of not bloccking the UI thread

Reply:
Thanks for pointing it out here. I added a comment about this tradeoff.